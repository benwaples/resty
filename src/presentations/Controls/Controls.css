#Controls {
  position: sticky;
  top: 50px;
  box-sizing: border-box;
  background-color: #8BAAAD ;
  form {
    align-self: flex-start;
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    width: 500px;

    label {
      margin: 5px;
      display: flex;
      flex-direction: column;
      justify-content: space-evenly;
      align-items: center;
      text-align: center;
      border: 1px solid black;
      padding: 8px;
      border-radius: 8px;
    }
  }
  
}