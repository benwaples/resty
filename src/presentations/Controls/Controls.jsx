import React from 'react';
import PropTypes from 'prop-types';

function Controls({
  url,
  method,
  body,
  onChange,
  onSubmit
}) {
  return (
    <div>
      <form onSubmit={onSubmit}>
        <label htmlFor="url">URL</label>
        <input 
          type="text"
          name="url"
          value={url}
          onChange={onChange}
        />
        <label htmlFor="method" >HTTP Request Method</label>
        <select name="method" onChange={onChange} value={method}>
          <option value="GET">GET</option>
          <option value="POST">POST</option>
          <option value="PUT">PUT</option>
          <option value="PATCH">PATCH</option>
          <option value="DELETE">DELETE</option>
        </select>
        <textarea name="body" onChange={onChange} value={body}></textarea>
      </form>
    </div>
  );
}

Controls.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Controls;

