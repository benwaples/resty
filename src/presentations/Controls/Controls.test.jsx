import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Controls from './Controls';

describe('Controls component', () => {
  afterEach(() => cleanup());

  it('should render controls', () => {
    const { asFragment } = render(<Controls 
      url="blank"
      method="GET"
      body="blank"
      onChange={() => {}}
      onSubmit={() => {}}
    />);

    expect(asFragment()).toMatchSnapshot(); 
  });

});

