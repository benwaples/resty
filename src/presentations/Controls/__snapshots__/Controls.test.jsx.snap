// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component should render controls 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="url"
      >
        URL
      </label>
      <input
        name="url"
        type="text"
        value="blank"
      />
      <label
        for="method"
      >
        HTTP Request Method
      </label>
      <select
        name="method"
      >
        <option
          value="GET"
        >
          GET
        </option>
        <option
          value="POST"
        >
          POST
        </option>
        <option
          value="PUT"
        >
          PUT
        </option>
        <option
          value="PATCH"
        >
          PATCH
        </option>
        <option
          value="DELETE"
        >
          DELETE
        </option>
      </select>
      <textarea>
        blank
      </textarea>
    </form>
  </div>
</DocumentFragment>
`;
